### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by PandaGrom**
# Description
[Add error handling for CRUD operations](https://trello.com/c/wGRbj1hx/70-add-error-handling-for-crud-operations)
## Type of change

- [ ] Bug fix
- [x] New feature 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/api/api.rb [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
   format :json
   prefix :api
   version 'v1', :path
 
+  use ErrorHandler
+
+  NON_AUTHENTICATION_PATHS = [
+    '/api/v1/sessions',
+    '/api/v1/registrations'
+  ].freeze
```
Comment (hannazspot) [inline app/api/api.rb]: could you pls review how to skip callbacks in controllers directly, but not in a general one:

there're some `skip_*` callbacks in rails => [documentation](https://api.rubyonrails.org/v6.1.4/classes/AbstractController/Callbacks/ClassMethods.html)

File: app/api/error_handler.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class ErrorHandler < Grape::Middleware::Base
+  def call!(env)
+    @env = env
+    begin
+      @app.call(@env)
+    rescue StandardError => e
```
Comment (hannazspot) [inline app/api/error_handler.rb]: I'd not catch all errors thru StandardError, it's too general. We'll lose visibility about errors that happen in our application.

So, I'd recommend catching exact errors: ActiveRecord::RecordNotFound, ActionController::ParameterMissing, ActiveModel::ValidationError

File: app/api/error_handler.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+      throw :error, message: e.message || options[:default_message], status: 500
```
Comment (hannazspot) [inline app/api/error_handler.rb]: could you please read about errors and why have to avoid raising 500 errors on our own

**Review by hannazspot — CHANGES_REQUESTED** [review]
The tests are supposed to be written when you add new logic

**Review by alhimick-prog — APPROVED** [review]

**Review by hannazspot — APPROVED** [review]

**Pull request merged**